pandas/core/aggregation.py

```diff
@@ -40,7 +43,9 @@ def is_multi_agg_with_relabel(**kwargs) -> bool:
     )
 
 
-def normalize_keyword_aggregation(kwargs: dict) -> Tuple[dict, List[str], List[int]]:
+def normalize_keyword_aggregation(
+    kwargs: dict,
+) -> Tuple[DefaultDict, Tuple[Hashable, ...], "np.ndarray"]:
```

Comment:
What issue is this fixing?

Reply:
see https://github.com/pandas-dev/pandas/pull/33263#discussion_r403133623

Reply:
@MomIsBestFriend can you git bisect or blame to find out what's the story behind the here.

Reply:
@simonjayhawkins It looks like no one can agree on the placement of this function, as it's moved around constantly.

---

But it looks like parts of the changes are made in #30497 while migrating from ```orderedict``` to ```defaultdict```

Reply:
normalize_keyword_aggregation is only called by aggregate in core\groupby\generic.py. 

the core\groupby\generic.py module is currently excluded by check_untyped_defs and aggregate does not have type hints.

maybe worth typing aggregate as well to ensure consistency.

Reply:
@simonjayhawkins Nice! thanks to that tip I could annotate with much more specific type hints.